addons/mail/models/mail_rtc_session.py

```diff
@@ -0,0 +1,141 @@
+        self.write({
+            'write_date': fields.Datetime.now(),
+        })
+
+    def update_and_broadcast(self, values):
+        if self.env.user.partner_id != self.partner_id:
+            return
+        self.write(values)
+        notifications = []
+        for member in self.channel_id.channel_last_seen_partner_ids:
+            notifications.append([
+                (self._cr.dbname, 'res.partner', member.partner_id.id),
+                {
+                    'type': 'rtc_session_data_update',
+                    'payload': {
+                        'rtcSession': self._mail_rtc_session_format()[0],
+                    },
+                },
+            ])
+        self.env['bus.bus'].sendmany(notifications)
+
+    @api.model
+    def _gc_inactive_sessions(self, max_age_minutes=3):
+        """ Garbage collect sessions that aren't active anymore,
+            this can happen when the server or the user's browser crash
+            or when the user's odoo session ends.
+        """
+        sessions = self.search([
+            ('write_date', '<', fields.Datetime.now() - relativedelta(minutes=max_age_minutes))
+        ])
+        if not sessions:
+            return
+        channel_ids = sessions.mapped('channel_id')
+        sessions.unlink()
+        channel_ids._notify_rtc_sessions_change()
+
+    def _disconnect(self):
+        """ Unlinks the sessions and notifies the associated partners that
+            their session ended.
+        """
+        notifications = []
+        for record in self:
+            notifications.append([
+                (self._cr.dbname, 'res.partner', record.partner_id.id),
+                {
+                    'type': 'rtc_session_ended',
+                    'payload': {
+                        'sessionId': record.id,
+                    },
+                },
+            ])
+        self.unlink()
+        return self.env['bus.bus'].sendmany(notifications)
+
+    def _notify_peers(self, targets, content):
+        """ Used for peer-to-peer communication,
+            guarantees that the sender is the current partner.
+
+            :param targets: a list of mail.rtc.session ids
+            :param content: a dict with the content to be sent to the targets
+        """
+
+        if not self.exists() or self.env.user.partner_id != self.partner_id:
+            return
+        notifications = []
+        target_sessions = self.search([('id', 'in', [int(target) for target in targets])])
+        for session in target_sessions:
+            notifications.append([
+                (self._cr.dbname, 'res.partner', session.partner_id.id),
+                {
+                    'type': 'rtc_peer_notification',
+                    'payload': {
+                        'sender': self.id,
+                        'content': content,
+                    },
+                },
+            ])
+        return self.env['bus.bus'].sendmany(notifications)
+
+    def _mail_rtc_session_format(self):
```

Comment:
I changed so we use the same format for both, In fact when looking at it, I didn't need to send the channel information at all since we either have it with channel_info (which is the main use of the by_channel), or the channel is already known (when updating the other fields).